src/librustc_codegen_ssa/back/write.rs

```diff
@@ -983,11 +980,13 @@ fn start_executing_work<B: ExtraBackendMethods>(
     // get tokens on `coordinator_receive` which will
     // get managed in the main loop below.
     let coordinator_send2 = coordinator_send.clone();
-    let helper = jobserver
-        .into_helper_thread(move |token| {
+    let helper = rustc_jobserver::helper_channel();
+    std::thread::spawn(move || {
```

Comment:
I think it would be best, if we can, to avoid this `std::thread::spawn`. I think that's relatively simple, though. The `request_token` method on the `Helper` structure can be updated to take a closure which is executed with the token, and wherever this file calls `request_token` it can instead use `request_token(|token| channel.send(token))` (or the equivalent thereof)

Reply:
Hm, are you thinking that we would stash these closures into some vector in the global struct and then call it from the jobserver crate's helper thread closure? I feel like that could work but is a bit odd; I'd rather pass the channel into the helper constructor (and say that we can have at most one at a time, basically).

Reply:
I don't think this'd touch the jobserver crate's helper thread, rustc already has its own helper coordination thread for doing this, and we'd just invoke the closures there. I'm basically trying to cut down on the number of "helper threads" we have here. Ideally one is too many, but right now I think we have at least 3 and would like to bring that number down.

Reply:
I think we can't block on the `request_token` call though? Maybe I'm missing something I guess...

Reply:
Well I don't think we have to necessarily pass around closures, but I'm pretty sure we can at least remove the need for a helper thread here. It's true that `request_token` can't block, but we could do something like pass `request_token` one end of a channel which is then handled in the closure of the global jobserver helper thread.